src/ToastNotification/TimedToastNotification.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import helpers from '../common/helpers'
+import Timer from '../common/Timer'
+import { ToastNotification } from '../index.js'
+
+/**
+ * TimedToastNotification Component for Patternfly React
+ */
+class TimedToastNotification extends React.Component {
+  constructor(props) {
+    super(props)
+    helpers.bindMethods(this, ['onMouseEnter', 'onMouseLeave'])
+  }
+
+  componentDidMount() {
+    const { onDismiss, timerDelay } = this.props
+
+    this.timer = new Timer(onDismiss, timerDelay)
+    this.timer.startTimer()
+  }
+
+  componentWillUnmount() {
+    this.timer && this.timer.clearTimer()
+  }
+
+  onMouseEnter() {
+    this.timer && this.timer.clearTimer()
+  }
+
+  onMouseLeave() {
+    this.timer && this.timer.startTimer()
+  }
+  render() {
+    const { notificationClass, type, onDismiss, children } = this.props
+    return (
+      <ToastNotification
+        notificationClass={notificationClass}
+        type={type}
+        onDismiss={onDismiss}
+      >
+        {children}
+      </ToastNotification>
+    )
+  }
+}
+
+TimedToastNotification.propTypes = {
+  /** timer delay */
+  timerDelay: PropTypes.number,
+  /** additional notification classes */
+  notificationClass: PropTypes.string,
+  /** callback when alert is dismissed  */
+  onDismiss: PropTypes.func,
+  /** the type of alert  */
+  type: PropTypes.oneOf(['danger', 'error', 'warning', 'success', 'info'])
+    .isRequired,
+  /** children nodes  */
+  children: PropTypes.node
+}
+TimedToastNotification.defaultProps = {
+  type: 'error',
```

Comment:
I think we did the same with `Alert`. Happy to change this if you want...I will just make Alert match the same.